### PR TITLE
Remove backwards compatibility in bulk app translations file

### PR DESCRIPTION
**Pull request by orangejenny**
This replaces https://github.com/dimagi/commcare-hq/pull/23870/ - merge conflicts were a bit hairy.

### CODE REVIEW

**Comment by orangejenny** [comment]
@mkangia I left https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/translations/integrations/transifex/parser.py#L61-L63 alone, wasn't 100% sure if it should be removed. Thoughts?

File: corehq/apps/translations/app_translations/upload_app.py [hunk 1]
```diff
@@ -267,7 +259,7 @@ def _check_for_sheet_warnings(app, sheet, headers):
 
 def _get_missing_cols(app, sheet, headers):
     expected_sheets = {h[0]: h[1] for h in headers}
-    expected_columns = _get_expected_headers(sheet, expected_sheets)
+    expected_headers = expected_sheets.get(sheet.worksheet.title, None)
     return set(expected_columns) - set(sheet.headers)
```
Comment (kaapstorm) [inline corehq/apps/translations/app_translations/upload_app.py]: Shouldn't this be
```python
    return set(expected_headers) - set(sheet.headers)
```
now?

**Comment by mkangia** [comment]
@orangejenny 
i made this change earlier, https://github.com/dimagi/commcare-hq/commit/df660c6415bc8578bb005dac332aba90dcd56375#diff-94a10c2ad54040c96cd01556311088d1

these things should be retained.

**Review by kaapstorm — APPROVED** [review]
Nice cleanup. Just the one comment.

File: corehq/apps/translations/app_translations/upload_app.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/translations/app_translations/upload_app.py]: (fixed in later commit, 4c01ce2ed9585367a7a2228b06f3126b1fcab582)

**Comment by orangejenny** [comment]
@mkangia thanks, I won't touch `parser.py` then

**Pull request merged**